enderio-machines/src/main/java/com/enderio/machines/common/blockentity/MachineState.java

```diff
@@ -45,4 +45,21 @@ public record MachineState(MachineStateType type, MutableComponent component) {
 
     public static final NetworkDataSlot.CodecType<Set<MachineState>> DATA_SLOT_TYPE
         = NetworkDataSlot.CodecType.createSet(CODEC, STREAM_CODEC.cast());
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
This is a record, so it shouldn't need these methods?

Reply:
yes, but by using the identity compare on the components we win some performance. This means we can only have 1 instance of the translation components, but that should be fine.